File: java/src/main/java/toolgood/algorithm/internals/functions/value/Function_ValueBoolean.java
package toolgood.algorithm.internals.functions.value;

import toolgood.algorithm.AlgorithmEngine;
import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;
import toolgood.algorithm.internals.functions.Function_0;

public final class Function_ValueBoolean extends Function_0 {
    private final boolean _value;

    public Function_ValueBoolean(boolean value) {
        this._value = value;
    }

    @Override
    public String Name() {
        return _value ? "True" : "False";
    }

    @Override
    public Operand Evaluate(AlgorithmEngine engine, java.util.function.BiFunction<AlgorithmEngine, String, Operand> tempParameter) throws Exception {
        return _value ? Operand.True : Operand.False;
    }

    @Override
    public OperandType GetResultType() {
        return OperandType.BOOLEAN;
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        stringBuilder.append(Name());
    }
}

File: JavaScript/src/Internals/Functions/Value/Function_ValueBoolean.js
import { Function_0 } from '../Function_0.js';
import { Operand } from '../../../Operand.js';

class Function_ValueBoolean extends Function_0 {
    constructor(value) {
        super();
        this._value = value;
    }

    get Name() {
        return this._value ? "True" : "False";
    }

    evaluate(engine, tempParameter = null) {
        return this._value ? Operand.True : Operand.False;
    }

    toString2(stringBuilder, addBrackets) {
        stringBuilder.push(this.Name);
    }
}

export { Function_ValueBoolean };